src/mcts/stoppers/factory.cc

```diff
@@ -162,7 +162,7 @@ void PopulateStoppers(ChainedSearchStopper* stopper, const OptionsDict& options,
     stopper->AddStopper(std::make_unique<DepthStopper>(*params.depth));
   }
 
-  PopulateStoppersForSelfplay(stopper, options);
+  if (!infinite) PopulateStoppersForSelfplay(stopper, options);
```

Comment:
Could you also add a comment on top of `PopulateStoppersForSelfplay` that it's not called in `uci infinite` mode (so that we don't add some stopper that is not suitable there)..
Also it feels that `PopulateStoppersForSelfplay` needs different name, but I cannot think of any..
"PopulateTreeBasedStoppers"? "PopulateSearchRelatedStoppers"?.. Those are stoppers that come from "inside" of search rather than external commands.

E.g. there's idea to also add "stoppers" which extend thinking sometimes (e.g. when move with top N doesn't have highest Q).

Reply:
"PopulateInternalStoppers"? "PopulateIntrinsicStoppers"?..

Reply:
Renamed to `PopulateIntrinsicStoppers` and added various comments.